### ISSUE TITLE
[peak_anno] Undefined chromosome error

### ISSUE THREAD

**Issue by Norisad** [dputhier/pygtftk#66]
I had an issue with the -z command, returning this error message.

```
 |-- 14:07-ERROR-peak_anno : Chromosome  i from GTF is undefined in /gpfs/tagc/home/sadouni/tf/split_tf/EOMES.bed file.
```

Quentin

**Comment by dputhier** [comment]
could you provide the other argument used.

**Comment by Norisad** [comment]
After testing, -z may not be the problem. Reusing an actual gtf did not solve it, what worked was using an actual chromsize file instead of `-c mm9`

**Comment by dputhier** [comment]
And when using -c mm9 and -K toto, could you paste the content of the chrom_size file produced ?

**Comment by dputhier** [comment]
I just checked -c mm9 with an example and could not see any problem. I guess it is something more subtle.

**Comment by dputhier** [comment]
What about the format of 

/gpfs/tagc/home/sadouni/tf/split_tf/EOMES.bed

It seems that the message is rather explicit telling that there is a chromosome named i in EOMES.bed. I agree this is weird but I would like to check. 
Could you provide the file as attached document ?

**Comment by dputhier** [comment]
And also the gtf used.

**Comment by dputhier** [comment]
Sorry. This i chromosome is related to a bug:

The line 

      message("Chromosome " + " i from GTF is undefined in --chrom-info file.",
                            type="ERROR")

should be 


message("Chromosome " + i + "  from GTF is undefined in --chrom-info file.",
                            type="ERROR")


I fixed it in ba205ddecff8b59564a57dde63a5dc491c167675

**Comment by dputhier** [comment]
Could you rerun and give the new error message plz.
Best

**Comment by qferre** [comment]
I can reproduce the bug with my own files : I have ran into this error again, independently, when trying to process random peaks generated from a human genome. I will use your fixed code and report back.

**Comment by qferre** [comment]
The fix in ba205ddecff8b59564a57dde63a5dc491c167675 was incomplete. I completed the fix in 4b6d6eaf5afed42055bd5af92c3c8e12a973cb61 on another line.

**Comment by qferre** [comment]
In my own test files, the problematic chromosome was "chr3_KI270779v1_alt". I assume the chromsizes you download by default do not include the "alt" ones ?

**Comment by dputhier** [comment]
Maybe we could add an argument so that the program may continue when a chromosome from the bed file is not define is the genome the program.

**Comment by dputhier** [comment]
I mean from the GTF file...

**Comment by qferre** [comment]
A possibility is to remove all lines whose first block (the chromosome) is not in the known chromosomes. Can this be done easily in pybedtools ?

**Comment by dputhier** [comment]
OK. We fixed the 'i' that was written instead of the corresponding chr name...
Moreover I added three novel arguments to silently delete features located on chromosomes unknown in --chrom-info file and found in (i) gtf files, (ii) peak files and (iii) more-bed files. 

	 -f, --force-chrom-gtf       Discard silently from GTF genes outside chromosomes defined in --chrom-info. (default: False)
	 -w, --force-chrom-peak      Discard silently from --peak-file peaks outside chromosomes defined in --chrom-info. (default: False)
	 -q, --force-chrom-more-bed  Discard silently from --more-bed files regions outside chromosomes defined in --chrom-info. (default: False)

This is implemented in 4356559f7f3bc95ad8e10df4f6112dca47f0b03a
